packages/loot-core/src/server/accounts/export-to-csv.ts

```diff
@@ -75,25 +77,55 @@ export async function exportQueryToCSV(query) {
       .options({ splits: 'all' }),
   );
 
-  const parentsPayees = new Map();
+  // initialize a map to allow splits to have correct number of split from
+  const parentsChildCount: Map<number, number> = new Map();
+  const childSplitOrder: Map<number, number> = new Map();
+
+  // first loop through and find parent transactions
   for (const trans of transactions) {
     if (trans.IsParent) {
-      parentsPayees.set(trans.Id, trans.Payee);
+      parentsChildCount.set(trans.Id, 0);
     }
   }
```

Comment:
What do you think about doing it all in one loop? Something like:

```suggestion
  for (const trans of transactions) {
    if (trans.IsChild) {
      let childNumber = parentsChildCount.get(trans.ParentId) || 0;
      childSplitOrder.set(trans.Id, childNumber);
      parentsChildCount.set(trans.ParentId, childNumber + 1);
    }
  }
```

Reply:
smart! i thought i needed two to make sure i had all the parents first, but i see here that i clearly don't, because it's a safe assumption that if it's a child it has a parent. 

incorporated these changes, down to one loop.